pgcontents/query.py

```diff
@@ -798,26 +807,34 @@ def _generate_notebooks(table, timestamp_column,
 
     # Decrypt each notebook and yield the result.
     for nb_row in result:
-        # The decrypt function depends on the user
-        user_id = nb_row['user_id']
-        decrypt_func = crypto_factory(user_id).decrypt
-
-        nb_dict = to_dict_with_content(table.c, nb_row, decrypt_func)
-        if table is files:
-            # Correct for files schema differing somewhat from checkpoints.
-            nb_dict['path'] = nb_dict['parent_name'] + nb_dict['name']
-            nb_dict['last_modified'] = nb_dict['created_at']
-
-        # For 'content', we use `reads_base64` directly. If the db content
-        # format is changed from base64, the decoding should be changed
-        # here as well.
-        yield {
-            'id': nb_dict['id'],
-            'user_id': user_id,
-            'path': to_api_path(nb_dict['path']),
-            'last_modified': nb_dict['last_modified'],
-            'content': reads_base64(nb_dict['content']),
-        }
+        try:
+            # The decrypt function depends on the user
+            user_id = nb_row['user_id']
+            decrypt_func = crypto_factory(user_id).decrypt
+
+            nb_dict = to_dict_with_content(table.c, nb_row, decrypt_func)
+            if table is files:
+                # Correct for files schema differing somewhat from checkpoints.
+                nb_dict['path'] = nb_dict['parent_name'] + nb_dict['name']
+                nb_dict['last_modified'] = nb_dict['created_at']
+
+            # For 'content', we use `reads_base64` directly. If the db content
+            # format is changed from base64, the decoding should be changed
+            # here as well.
+            yield {
+                'id': nb_dict['id'],
+                'user_id': user_id,
+                'path': to_api_path(nb_dict['path']),
+                'last_modified': nb_dict['last_modified'],
+                'content': reads_base64(nb_dict['content']),
+            }
+        except CorruptedFile:
+            if logger is not None:
+                logger.warning(
+                    'Corrupted file with id {} in table {}.',
```

Comment:
The standard library `logging` module uses printf-style format strings (i.e. `%s` instead of `{}`).  We use `logbook` for most quantopian projects, but I don't think we have a dependency on it here.  The easiest workaround here is probably just to do the string formatting here ourselves.  Normally that's a bad practice for logging, because one of the benefits of logging is that you can avoid the cost of string formatting if you're logging at a level you don't care about, but here the logger is already optional, so I think it's reasonable to expect that the consumer is going to format these values eventually.